### ISSUE TITLE
asPaginator keeps returning the same results

### ISSUE THREAD

**Issue by gajus** [PLhery/node-twitter-api-v2#116]
```ts
const twitter = new Twitter(argv['twitter-bearer-token']).readOnly.v2;

const user = (await twitter.userByUsername('gajus')).data;

const followerPaginator = await twitter.followers(user.id, {
  asPaginator: true,
  max_results: 1_000,
});

for await (const follower of followerPaginator) {
  console.log('follower', follower);
}
```

Returns

```
follower { id: '1080856485528244226', name: 'ben', username: '_BenHQ' }
follower { id: '288845518', name: 'aleida cray', username: 'aleidaipcray' }
follower { id: '14738889', name: 'DrunkenDuckJapan', username: 'Mr1Duck' }
follower { id: '1080856485528244226', name: 'ben', username: '_BenHQ' }
follower { id: '288845518', name: 'aleida cray', username: 'aleidaipcray' }
follower { id: '14738889', name: 'DrunkenDuckJapan', username: 'Mr1Duck' }
follower { id: '1080856485528244226', name: 'ben', username: '_BenHQ' }
follower { id: '288845518', name: 'aleida cray', username: 'aleidaipcray' }
follower { id: '14738889', name: 'DrunkenDuckJapan', username: 'Mr1Duck' }
follower { id: '1080856485528244226', name: 'ben', username: '_BenHQ' }
follower { id: '288845518', name: 'aleida cray', username: 'aleidaipcray' }
follower { id: '14738889', name: 'DrunkenDuckJapan', username: 'Mr1Duck' }
follower { id: '1080856485528244226', name: 'ben', username: '_BenHQ' }
follower { id: '288845518', name: 'aleida cray', username: 'aleidaipcray' }
follower { id: '14738889', name: 'DrunkenDuckJapan', username: 'Mr1Duck' }
[..]
```

Expected:

```
follower { id: '1080856485528244226', name: 'ben', username: '_BenHQ' }
follower { id: '288845518', name: 'aleida cray', username: 'aleidaipcray' }
follower { id: '14738889', name: 'DrunkenDuckJapan', username: 'Mr1Duck' }
```

**Comment by gajus** [comment]
Also, is this thread still accurate?

https://github.com/PLhery/node-twitter-api-v2/issues/81

Reading documentation, I thought that paginator automatically handles rate limit.

**Comment by gauloics** [comment]
got TIMEOUT for me, help please

**Comment by alkihis** [comment]
Hello, sorry for the long delay.
I've identified the encountered issue.
It happens with paginators having exactly one page, only in Twitter v2 API and only when you use the `for await-of` iterator.

For your second question, it does when using `for await-of` or `.fetchLast`. It automatically stops fetching (without any error thrown), but it does not prevent you to call `.next`.

**Comment by alkihis** [comment]
Hi again,

After running an `npm update`, do you still experiment issues with paginators?
It should be fixed since 1.7.2 :)
